### DISCUSSION TITLE
GlobalMiddlewares are called once before each @Field

### DISCUSSION THREAD

**Discussion by AmrAnwar** [MichalLytek/type-graphql#902] [Q&A] [answered]
as I understand GlobalMiddlewares run before each @query, @mutation, @fieldResolver, @Subscription, but do it run also before each @Field in the ObjectType when we return its fields? 

example for that :

resolver:
```typescript

@ObjectType()
class TestObject {
	@Field()
	test: string;

	@Field()
	test2: string;
}

@Resolver((of) => TestObject)
export class TestResolver {
	@Query((returns) => TestObject)
	async testResolver(): Promise<TestObject> {
		console.log('TestResolver...1');
		return <TestObject><unknown>{ test1: "test1", test2: "test2" };
	}
}

```

middleware:
```typescript
	getUserMiddleware: async ({ context: { request } }: any, next: any) => {
		console.log("getUserMiddleware....")
		return next();
	},

```
schema:
```typescript
await buildSchema({
		resolvers: [TestResolver],
		globalMiddlewares: [getUserMiddleware],
	});
```

test code:
I get the log from middleware 3 times, once at the beginning before the query function and two times at the end for each field.
![MicrosoftTeams-image (10)](https://user-images.githubusercontent.com/25433336/120327075-a963f180-c2e9-11eb-8bd1-defee8029009.png)

	
------

### if it's so, is there a way to prevent that ? and thanks

typegraphql version: "^1.1.1"

**Comment by MichalLytek** [comment, marked as answer]
Yes, global middlewares runs on every graphql resolver - query, mutation, field.

There's no API for limiting that, you can only check `info` and quick exit on fields if you don't want that behavior.

**Reply by AmrAnwar** [reply]
thanks yeah I'll add that
